app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -70,25 +70,57 @@ class HikesViewModel(
 
   private var _hikeFlowsMap = mutableMapOf<String, MutableStateFlow<Hike>>()
 
+  private val _osmDataRetrievalMutex = Mutex()
+
+  private val _allOsmDataLoaded = MutableStateFlow(true)
+
+  private val _loadingMutex = Mutex()
+
+  private var _ongoingLoadingOperations: Int = 0
+
   private val _loading = MutableStateFlow(false)
 
   private val _hikeFlowsList = MutableStateFlow<List<StateFlow<Hike>>>(emptyList())
 
   private val _selectedHike = MutableStateFlow<Hike?>(null)
 
-  /** Internal type used to store where the hikes in [_hikeFlowsMap] come from. */
-  private enum class LoadedHikes {
+  /**
+   * Enum used to designate where the hikes loaded in [hikeFlows] come from semantically.
+   *
+   * See the possible values for this enum along with [loadedHikesType] for more information.
+   */
+  enum class LoadedHikes {
```

Comment:
Answer: I am guessing the reason is testing, after looking at the tests.

Reply:
Actually it's mostly because of `loadedHikesType`. In my previous implementation of `HikesViewModel`, this enum was used for an internal state, and it did not need to be known outside.

However I realized this state would be useful outside of `HikesViewModel`, so I created the state flow `loadedHikesType`.